velox/expression/Expr.cpp

```diff
@@ -361,17 +375,27 @@ void Expr::eval(
   // all the time. Therefore, we should delay loading lazy vectors until we
   // know the minimum subset of rows needed to be loaded.
   //
+  // Load fields multiple referenced by inputs unconditionally. It's hard to
+  // know the superset of rows the multiple inputs need to load.
+  //
   // If there is only one field, load it unconditionally. The very first IF,
   // AND or OR will have to load it anyway. Pre-loading enables peeling of
   // encodings at a higher level in the expression tree and avoids repeated
   // peeling and wrapping in the sub-nodes.
   //
-  // TODO: Re-work the logic of deciding when to load which field.
+  // TODO: only pre-loading lazy vectors that is not flat encoding,
```

Comment:
I don't understand this TODO. Would you explain what it means? Is it necessary to change the original TODO?

Reply:
I revert my change to this comment. And I will open another issue to discuss the conflict between delay loading and peeling of encodings.